### PR TITLE
Implement `petab.*` deprecation with `__getattr__`

### PR DESCRIPTION
**Pull request by dilpath**

### CODE REVIEW

File: petab/v1/problem.py [hunk 1]
```diff
@@ -79,6 +79,13 @@ def __init__(
         mapping_df: pd.DataFrame = None,
         extensions_config: dict = None,
     ):
+        if getattr(self, "deprecated_use", False):
+            warn(
+                "It looks like you are using the `Problem` class like "
+                "`from petab import Problem`. Please change to "
+                "`from petab.v1 import Problem`.",
```
Comment (dweindl) [inline petab/v1/problem.py:86]:
```suggestion
                "`from petab.v1 import Problem`.", DeprecationWarning
```

**Review by dweindl — COMMENTED** [review]
Thx. It will break any `isinstance(petab.v1.Problem(...), petab.Problem)` not sure if that's a real concern?!

File: petab/__init__.py [hunk 2]
```diff
@@ -2,74 +2,26 @@
...
-# deprecated imports
-from petab.v1 import *  # noqa: F403, F401, E402
-
-from .v1.format_version import __format_version__  # noqa: F401, E402
-
-# __all__ = [
-#     'ENV_NUM_THREADS',
-# ]
-
 ENV_NUM_THREADS = "PETAB_NUM_THREADS"
```
Comment (dilpath) [inline petab/__init__.py:14]:
```suggestion
ENV_NUM_THREADS = "PETAB_NUM_THREADS"
__all__ = "ENV_NUM_THREADS"
```

**Comment by dilpath** [comment]
Should `from .versions import *` be in `petab/__init__.py`?

**Comment by dweindl** [comment]
> Should `from .versions import *` be in `petab/__init__.py`?

I am undecided what we should import there.

**Pull request merged**